lib/facility.rb

```diff
@@ -1,14 +1,73 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility)
+    @name = facility[:name]
+    @address = facility[:address]
+    @phone = facility[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    
+    if @services.include?('Vehicle Registration') == true
+      @registered_vehicles << vehicle
+      vehicle.set_registration_date
+      vehicle.set_plate_type
+
+      if vehicle.antique? == true
+        @collected_fees += 25
+      elsif vehicle.electric_vehicle? == true
+        @collected_fees += 200
+      else
+        @collected_fees += 100
+      end
+    end
+
+    if @registered_vehicles.empty? == true
```

Comment:
`@registered_vehicles.empty?` will evaluate to true, so checking that it is `== true` is redundant. Same could be said for your elsif before